shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/interactor/AnalyticInteractor.kt

```diff
@@ -43,58 +27,34 @@ class AnalyticInteractor(
     }
 
     override suspend fun flushEvents() {
-        launchHyperskillFlushEventsJob(withDelay = false)
+        hyperskillAnalyticEngine.flushEvents()
+        appsFlyerAnalyticEngine?.flushEvents()
+
         eventMonitor?.analyticDidFlushEvents()
     }
 
     suspend fun logEvent(event: AnalyticEvent, forceLogEvent: Boolean = false) {
         when (event.source) {
-            AnalyticSource.HYPERSKILL_API -> logHyperskillEvent(event, forceLogEvent)
+            AnalyticSource.HYPERSKILL_API -> {
+                hyperskillAnalyticEngine.reportEvent(event, force = forceLogEvent)
+            }
+            AnalyticSource.APPS_FLYER -> {
+                appsFlyerAnalyticEngine?.reportEvent(event, force = forceLogEvent)
+            }
         }
         eventMonitor?.analyticDidReportEvent(event)
     }
 
     override fun setScreenOrientation(screenOrientation: ScreenOrientation) {
-        this.screenOrientation = screenOrientation
-    }
-
-    private suspend fun logHyperskillEvent(event: AnalyticEvent, forceLogEvent: Boolean) {
-        kotlin.runCatching {
-            if (event !is HyperskillAnalyticEvent) {
-                return
-            }
-
-            // Prevent multiple network calls when no cached profile
-            val currentProfile: Profile =
-                currentProfileStateRepository.getState().getOrElse { return }
-
-            val processedEvent = hyperskillEventProcessor.processEvent(
-                event = event,
-                userId = currentProfile.id,
-                isNotificationsPermissionGranted = notificationInteractor.isNotificationsPermissionGranted(),
-                screenOrientation = screenOrientation ?: ScreenOrientation.PORTRAIT
-            )
-            hyperskillRepository.logEvent(processedEvent)
-
-            if (flushEventsJob != null && !flushEventsJob!!.isCompleted) {
-                return
-            }
-
-            launchHyperskillFlushEventsJob(withDelay = !forceLogEvent)
+        // TODO: Find a better way for providing ScreenOrientation
+        if (hyperskillAnalyticEngine is HyperskillAnalyticEngine) {
+            hyperskillAnalyticEngine.setScreenOrientation(screenOrientation)
         }
     }
 
-    private suspend fun launchHyperskillFlushEventsJob(withDelay: Boolean) {
-        flushEventsJob?.cancelAndJoin()
-        flushEventsJob = MainScope().launch {
-            if (withDelay) {
-                delay(FLUSH_EVENTS_DELAY_DURATION)
-            }
-
-            val isAuthorized = authInteractor.isAuthorized()
-                .getOrDefault(false)
-
-            hyperskillRepository.flushEvents(isAuthorized)
+    override fun setAppTrackingTransparencyAuthorizationStatus(isAuthorized: Boolean) {
+        if (hyperskillAnalyticEngine is HyperskillAnalyticEngine) {
+            hyperskillAnalyticEngine.setAppTrackingTransparencyAuthorizationStatus(isAuthorized = isAuthorized)
         }
     }
```

Comment:
In my opinion, it's better to not expose the internals of the engine to the Analytics interaction.
We can add `setAppTrackingTransparencyAuthorizationStatus` to the `AnalyticEngine` interface. 
And only Hyperskill implementation will do some work.
And I think, we can do the same with `setScreenOrientation `.